lib/posix-process/signal/signal.c

```diff
@@ -230,6 +231,9 @@ int pprocess_signal_pdesc_init(struct posix_process *process)
 	for (__sz i = 0; i < SIG_ARRAY_COUNT; i++)
 		pprocess_signal_sigaction_clear(&pd->sigaction[i]);
 
+	uk_mutex_init(&pd->lock);
```

Comment:
Should we maybe make this mutex recursive? `UK_MUTEX_CONFIG_RECURSE`

Reply:
Same comment regarding mutexes as above, will revise in a follow up iteration.